### ISSUE TITLE
Feature Request: Support dates with negative years (i.e. BCE)

### ISSUE THREAD

**Issue by fdw** [moment/luxon#774]
I have tried to create `DateTime` objects in BCE, for example like this `DateTime.parseISO('-7000-0-0')` or `DateTime.fromFormat('-7000', 'yyyyy'). However, they all have `NaN` as year.

How can I create such dates?

**Comment by balagge** [comment]
Does not seem to be supported. If you check the `invalid` property of the returned object, you get

`Invalid { reason: 'unparsable', explanation: 'the input "-7000" can't be parsed as format yyyyy' }`,

which means negative sign is not parsed with format token `yyyyy`.

Documentation does not help too much:

- doc of [fromFormat](https://moment.github.io/luxon/docs/class/src/datetime.js~DateTime.html#static-method-fromFormat) refers to a linked table for supported tokens;
- the [linked table](https://moment.github.io/luxon/docs/manual/parsing.html#table-of-tokens) contains several year tokens, `y`, `yy`, `yyyy`, `yyyyy`, `yyyyyy`, but the description never says anything about sign.

I did some debugs. The parsing is quite complex, but the `yyyyy` token will construct the regexp `/^(\d{4,6})$/i`, which is used to parse the input. See `function explainFromTokens(locale, input, format)` in `src/impl/tokenParser.js` for details - the important part of the regexp is built by `function unitForToken(token, loc)`. All the other year tokens will result in similar regexps.

Bottom line: it seems only digits are accepted for years. A feature request seems feasible.

**Comment by fdw** [comment]
That's a pity; moment.js supported it with `y`. But thank you for doing all the digging!

Can I just update this issue to a feature request, or should I close and open a new one?

**Comment by balagge** [comment]
I am just a user here, and I am not aware of the procedures. Normally I would choose updating. I like to have the complete history in a ticket.

**Comment by fdw** [comment]
Sorry, I thought you work here ;)
Thanks for your help!

**Comment by icambron** [comment]
This just isn't how ISO 8601 works, y'all. To use `-`, you need to provide a 6-digit year, like so:

```js
DateTime.fromISO("-007000-01-01")
```

That's just what the spec says.

It's true that the format parser doesn't support negative years for `yyyy`, `yyyyy`, or `yyyyyy`. I'm not sure that's a terrible thing, but I'd probably accept a PR for it.

**Comment by fdw** [comment]
Thank you, support in `fromISO` is enough for me. I just hadn't seen the necessity of six digits before.

**Comment by GillesDebunne** [comment]
For reference, `DateTime.fromObject({year:-1234, month:9, day:25})` is also an option.

**Comment by 59023g** [comment]
I don't see a way to include year notation, ie `BCE`, `AD` when I invoke `toLocaleString()`. Does this exist?

**Comment by dasa** [comment]
Try: `luxon.DateTime.fromObject({year:-1234, month:9, day:25}).toLocaleString({ ...luxon.DateTime.DATE_FULL, era: "short" })` -> "September 24, 1235 BC"

See "era" options at: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat
